pkg/model/openstackmodel/servergroup.go

```diff
@@ -268,24 +268,36 @@ func (b *ServerGroupModelBuilder) Build(c *fi.CloudupModelBuilderContext) error
 		} else {
 			affinityPolicies = append(affinityPolicies, "anti-affinity")
 		}
-		sgTask := &openstacktasks.ServerGroup{
-			Name:        s(fmt.Sprintf("%s-%s", clusterName, ig.Name)),
-			ClusterName: s(clusterName),
-			IGName:      s(ig.Name),
-			Policies:    affinityPolicies,
-			Lifecycle:   b.Lifecycle,
-			MaxSize:     ig.Spec.MaxSize,
+
+		sgName := fmt.Sprintf("%s-%s", clusterName, ig.Name)
+		if name, ok := ig.ObjectMeta.Annotations[openstack.OS_ANNOTATION+openstack.SERVER_GROUP_NAME]; ok {
+			sgName = fmt.Sprintf("%s-%s", clusterName, name)
+		}
+
+		sgTask, ok := sgs[sgName]
+		if !ok {
+			igMap := make(map[string]*int32)
+			igMap[ig.Name] = ig.Spec.MaxSize
+			sgTask = &openstacktasks.ServerGroup{
+				Name:        s(sgName),
+				ClusterName: s(clusterName),
+				IGMap:       igMap,
```

Comment:
the problem with this was that earlier one servergroup was mapped to one instancegroup. However, now when its possible to customise the servergroup name, it can be that multiple instancegroups are mapped to single servergroup. It makes the logic little bit advanced.